STIPS/STIP-009.md

```diff
@@ -0,0 +1,1178 @@
+
+- Allow `owner` to add and remove global `operator` permissions on extensions
+- Allow `owner` to limit `operator`(s) functionality on extensions with an asset whitelist
+- Allow `owner` to update asset whitelist
+- Allow `owner` to perform Set Token admin functions such as `addModule`, `removeModule`, and `setManager`
+- Allow `owner` to update ownerFeeSplit
+- Allow `owner` to update ownerFeeRecipient
+- Allow `owner` to transfer tokens held by the BaseManager to another address (sweeper)
+- Allow `owner` to update useAllowedAssetList flag. (When false, manager can trade any asset)
+- Allow extensions to interact with modules
+
+### TradeExtension
+
+- Allow `owner` to enable functionality of TradeModule with only a state change and no contract deployment
+- Allow privileged `operator`(s) to perform trades on a DEX
+- Allow `owner` to restrict assets the privileged `operator`(s) can trade into with an asset whitelist
+
+### BasicIssuanceExtension
+
+- Allow `owner` to enable functionality of BasicIssuanceModule with only a state change and no contract deployment
+- Allow `owner` and `methodologist` to split issuance and redemption fees
+- Allow `owner` to update the issuance and redemption fee
+- Allow `owner` to update the issuance and redemption fee recipient
+
+### StreamingFeeSplitExtension
+
+- Allow `owner` to enable functionality of StreamingFeeModule with only a state change and no contract deployment
+- Allow `owner` and `methodologist` to split streaming fees
+- Allow `owner` to update the streaming fee
+- Allow `owner` to update the streaming fee recipient
+
+## User Flows
+
+### DelegatedManagerFactory.createSetAndManager()
+
+![DelegatedManagerFactory create](../assets/stip-009/image2.png "")
+
+A `deployer` wants to create a new Set Token with a DelegatedManager smart contract manager.
+
+1. The `deployer` calls createSetAndManager() passing in parameters to create a Set Token, parameters for the permissioning on DelegatedManager, and the desired extensions. Specifically,
+
+    - components: List of addresses of components for initial positions
+    - units: List of units for initial positions
+    - name: Name of SetToken
+    - symbol: Symbol of SetToken
+    - owner: The address of the `owner`
+    - methodologist: The address of the `methodologist`
+    - operators: List of addresses of the `operator`(s)
+    - modules: List of addresses of modules to be enabled
+    - assets: List of addresses of assets for initial asset whitelist
+    - extensions: List of addresses of global extensions to be enabled
+
+2. Creation Parameters are validated:
+    - If assets are defined, asset list must match components
+
+3. A Set Token is deployed using SetTokenCreator
+
+4. A DelegatedManager is deployed with the DelegatedManagerFactory as the temporary `owner` until after initialization
+    - If assets are defined, constructor param *useAssetAllowList* is set to true, false otherwise.
```

Comment:
Not as concerned about it in this STIP but lets make sure we align on the camel case for allowlist, is it `assetAllowlist` or `assetAllowList` when we're writing the code. I've been using `assetAllowlist`